### ISSUE TITLE
Windows / MinGW test errors

### ISSUE THREAD

**Issue by MSP-Greg** [eventmachine/eventmachine#844]
Today was testing EM, both on Appveyor (my EM fork) and locally.  AV testing locked, local testing completed, but had four errors, two in `test_attach.rb`, and two in `test_timers.rb`.

Below is a log with details.  Be happy to test any changes...

Thanks, Greg

```
ruby 2.6.0dev (2018-05-06 trunk 63343) [x64-mingw32]

EM Library Type: extension
no IPv6 in this host, skipping tests in C:/GitHub/eventmachine/tests/test_ipv6.rb
EM.popen not implemented, skipping tests in C:/GitHub/eventmachine/tests/test_processes.rb
OPENSSL_LIBRARY_VERSION: OpenSSL 1.1.0h  27 Mar 2018
        OPENSSL_VERSION: OpenSSL 1.1.0h  27 Mar 2018
              SSL_AVAIL: tlsv1 tlsv1_1 tlsv1_2

276 tests, 3524 assertions, 0 failures, 4 errors, 6 pendings, 13 omissions, 0 notifications

 1) C:/GitHub/eventmachine/tests/test_attach.rb:181:in `test_attach_data'
     178:     $watch_only = nil
     179:     $read = []
     180:
  => 181:     EM.run do
     182:       EM.attach r do |c|
     183:         assert_false(c.watch_only?)
     184:         def c.receive_data data
C:/GitHub/eventmachine/lib/eventmachine.rb:196:in `run'
C:/GitHub/eventmachine/lib/eventmachine.rb:196:in `run_machine'
C:/GitHub/eventmachine/tests/test_attach.rb:182:in `block in test_attach_data'
C:/GitHub/eventmachine/lib/eventmachine.rb:743:in `attach'
C:/GitHub/eventmachine/lib/eventmachine.rb:773:in `attach_io'
C:/GitHub/eventmachine/tests/test_attach.rb:183:in `block (2 levels) in test_attach_data'
C:/GitHub/eventmachine/lib/em/connection.rb:772:in `watch_only?'
Error: test_attach_data(TestAttach):
  NoMethodError: undefined method `watch_only?' for EventMachine:Module
  Did you mean?  watch_file

 2) C:/GitHub/eventmachine/tests/test_attach.rb:157:in `test_watch_only'
     154:     r, w = IO.pipe
     155:     $watch_only = nil
     156:
  => 157:     EM.run do
     158:       EM.watch r do |c|
     159:         assert_true(c.watch_only?)
     160:         c.notify_readable = true
C:/GitHub/eventmachine/lib/eventmachine.rb:196:in `run'
C:/GitHub/eventmachine/lib/eventmachine.rb:196:in `run_machine'
C:/GitHub/eventmachine/tests/test_attach.rb:158:in `block in test_watch_only'
C:/GitHub/eventmachine/lib/eventmachine.rb:733:in `watch'
C:/GitHub/eventmachine/lib/eventmachine.rb:773:in `attach_io'
C:/GitHub/eventmachine/tests/test_attach.rb:159:in `block (2 levels) in test_watch_only'
C:/GitHub/eventmachine/lib/em/connection.rb:772:in `watch_only?'
Error: test_watch_only(TestAttach):
  NoMethodError: undefined method `watch_only?' for EventMachine:Module
  Did you mean?  watch_file

 7) C:/GitHub/eventmachine/tests/test_timers.rb:105:in `test_add_timer_increments_timer_count'
     102:   end
     103:
     104:   def test_add_timer_increments_timer_count
  => 105:     EM.run {
     106:       n = EM.get_timer_count
     107:       EM::Timer.new(0.01) {
     108:         EM.stop
C:/GitHub/eventmachine/lib/eventmachine.rb:196:in `run'
C:/GitHub/eventmachine/lib/eventmachine.rb:196:in `run_machine'
C:/GitHub/eventmachine/tests/test_timers.rb:106:in `block in test_add_timer_increments_timer_count'
Error: test_add_timer_increments_timer_count(TestTimers):
  NoMethodError: undefined method `get_timer_count' for EventMachine:Module
  Did you mean?  get_max_timer_count

 8) C:/GitHub/eventmachine/tests/test_timers.rb:115:in `test_timer_run_decrements_timer_count'
     112:   end
     113:
     114:   def test_timer_run_decrements_timer_count
  => 115:     EM.run {
     116:       n = EM.get_timer_count
     117:       EM::Timer.new(0.01) {
     118:         assert_equal(n, EM.get_timer_count)
C:/GitHub/eventmachine/lib/eventmachine.rb:196:in `run'
C:/GitHub/eventmachine/lib/eventmachine.rb:196:in `run_machine'
C:/GitHub/eventmachine/tests/test_timers.rb:116:in `block in test_timer_run_decrements_timer_count'
Error: test_timer_run_decrements_timer_count(TestTimers):
  NoMethodError: undefined method `get_timer_count' for EventMachine:Module
  Did you mean?  get_max_timer_count
```

**Comment by MSP-Greg** [comment]
1\. I checked, and several of the other module functions defined in `rubymain.cpp` exist.

2\. I bypassed the four above tests, and testing Ruby 2.0 thru trunk works fairly well, except for consistent errors in 32 bit 2.4.3 and 2.5.0.  Appveyor should be updating these Ruby versions shortly.

Results are at https://ci.appveyor.com/project/MSP-Greg/eventmachine/build/9.

All versions are compiled with OpenSSL (4 different packages), and I moved the OpenSSL package code to a PowerShell file.  Previously, the appveyor.yml file used proprietary Appveyor code to download the packages, the new PowerShell script does not.

Greg
